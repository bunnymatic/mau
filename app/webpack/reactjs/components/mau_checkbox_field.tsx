import { MauHint } from "@reactjs/components/mau_hint";
import cx from "classnames";
import { Field } from "formik";
import React, { FC, JSX } from "react";

interface MauCheckboxFieldProps {
  name: string;
  label: string;
  hint?: string | JSX.Element;
  id?: string;
  classes?: string;
  disabled?: boolean;
}

export const MauCheckboxField: FC<MauCheckboxFieldProps> = ({
  id,
  name,
  label,
  hint,
  classes,
  disabled,
}) => {
  return (
    <div
      className={cx("boolean input optional", { [classes]: Boolean(classes) })}
    >
      <label htmlFor={name}>
        <Field
          type="checkbox"
          name={name}
          id={id ?? name}
          disabled={disabled}
        />
        <span>{label}</span>
      </label>
      <MauHint>{hint}</MauHint>
    </div>
  );
};
