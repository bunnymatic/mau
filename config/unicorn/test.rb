# frozen_string_literal: true

# Bind to 127.0.0.1 to avoid issue with MacOS and `getaddrinfo`
# causing ruby crashes during asset compilation and startup.
# https://bugs.ruby-lang.org/issues/15763
listen '127.0.0.1:3000'
