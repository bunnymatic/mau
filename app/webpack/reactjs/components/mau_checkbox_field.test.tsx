import { renderInForm } from "@reactjs/test/renderers";
import React from "react";
import { describe, expect, it } from "vitest";

import { MauCheckboxField } from "./mau_checkbox_field";

describe("MauCheckboxField", () => {
  it("matches the snapshot", () => {
    const { container } = renderInForm(
      <MauCheckboxField label="The Label" name="the_name" />
    );
    expect(container).toMatchSnapshot();
  });
});
