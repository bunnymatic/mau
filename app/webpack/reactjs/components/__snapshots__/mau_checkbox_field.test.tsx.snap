// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MauCheckboxField > matches the snapshot 1`] = `
<div>
  <div
    class="boolean input optional"
  >
    <label
      for="the_name"
    >
      <input
        id="the_name"
        name="the_name"
        type="checkbox"
        value=""
      />
      <span>
        The Label
      </span>
    </label>
  </div>
</div>
`;
